code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -16,12 +16,15 @@
 		return
 	user.dna = chosen_dna.Clone()
 	user.real_name = chosen_dna.real_name
-	user.flavor_text = ""
 	if(ishuman(user))
-		user.set_species()
+		user.set_species(chosen_dna.species)
+	domutcheck(user, null, MUTCHK_FORCED) //Ensures species that get powers by the species proc handle_dna keep them
+	user.flavor_text = ""
+	user.dna.UpdateSE()
+	user.dna.UpdateUI()
+	user.sync_organ_dna(1)
 	user.UpdateAppearance()
-	domutcheck(user, null)
-	
+
```

Comment:
For consistency with the other changes I made and to bring this in-line with how cloning handles giving people their identities and such

They're only slightly related, so I'll test again with these changes undone